libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -821,16 +843,44 @@ unsigned CompilerUtils::loadFromMemoryHelper(Type const& _type, bool _fromCallda
 	return numBytes;
 }
 
-void CompilerUtils::cleanHigherOrderBits(IntegerType const& _typeOnStack)
+void CompilerUtils::cleanHigherOrderBits(Type const& _typeOnStack)
 {
-	if (_typeOnStack.numBits() == 256)
-		return;
-	else if (_typeOnStack.isSigned())
-		m_context << u256(_typeOnStack.numBits() / 8 - 1) << Instruction::SIGNEXTEND;
+	auto cleanBits = [this](int const numBits, bool const isSigned) {
+		if (numBits == 256)
+			return;
+		else if (isSigned)
+			m_context << u256(numBits / 8 - 1) << Instruction::SIGNEXTEND;
+		else
+			m_context << ((u256(1) << numBits) - 1) << Instruction::AND;
+	};
+
+	if (auto intType = dynamic_cast<IntegerType const*>(&_typeOnStack)) 
+		cleanBits(intType->numBits(), intType->isSigned());
+	else if (auto fixedType = dynamic_cast<FixedPointType const*>(&_typeOnStack)) 
+		cleanBits(fixedType->numBits(), fixedType->isSigned());
 	else
-		m_context << ((u256(1) << _typeOnStack.numBits()) - 1) << Instruction::AND;
+		BOOST_THROW_EXCEPTION(
+			CompilerError() << errinfo_comment("Invalid type passed in too have higher bits cleaned.")
+		);
 }
 
+void CompilerUtils::shiftFixedPointNumber(FixedPointType const& _stackType, FixedPointType const& _targetType)
```

Comment:
Internally it is performing a shift, but this method is not used by the high level shifting in the language. For that it would need LValue and RValue?